viral_seq/analysis/get_features.py

```diff
@@ -55,6 +58,58 @@ def get_kmers(records, k=10, kmer_type="AA"):
     return kmers
 
 
+def get_bert_embeddings(records):
+    print("start of get_bert_embeddings")
+    MODEL_NAME = "Rostlab/prot_bert_bfd"
+    tokenizer = BertTokenizer.from_pretrained(MODEL_NAME, do_lower_case=False)
+    model = BertModel.from_pretrained(MODEL_NAME)
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    model = model.to(device)
+    model.eval()
+    protein_segments = 0
+    protein_embeddings_list = []
+    for record in records:
+        print("record:", record)
+        for feature in record.features:
+            if feature.type == "CDS":
+                nuc_seq = feature.location.extract(record.seq)
+                if len(nuc_seq) % 3 != 0:
+                    # bad cds are skipped as in
+                    # https://github.com/Nardus/zoonotic_rank/blob/main/Utils/GenomeFeatures.py#L105
+                    continue
+                this_seq = nuc_seq.translate()
+                print("this_seq:", this_seq)
+                # ProtBERT expects spaces between amino acids
+                sequence_w_spaces = " ".join(re.sub(r"[\\*~]", "", str(this_seq).upper()))
+                print("sequence_w_spaces:", sequence_w_spaces)
+                inputs = tokenizer(
+                        sequence_w_spaces,
+                        padding="max_length",
+                        max_length=2048,
```

Comment:
need a smaller truncation than this for ESM? may want to consider that in the design/flexibility here if we decide to generalize to allow ESM as well